activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -442,16 +442,12 @@ def pk_and_sequence_for(table) #:nodoc:
 
         def primary_keys(table_name) # :nodoc:
           select_values(<<-SQL.strip_heredoc, "SCHEMA")
-            WITH pk_constraint AS (
-              SELECT conrelid, unnest(conkey) AS connum FROM pg_constraint
-              WHERE contype = 'p'
-                AND conrelid = #{quote(quote_table_name(table_name))}::regclass
-            ), cons AS (
-              SELECT conrelid, connum, row_number() OVER() AS rownum FROM pk_constraint
-            )
-            SELECT attr.attname FROM pg_attribute attr
-            INNER JOIN cons ON attr.attrelid = cons.conrelid AND attr.attnum = cons.connum
-            ORDER BY cons.rownum
+            SELECT a.attname, format_type(a.atttypid, a.atttypmod) AS data_type
+            FROM   pg_index i
+            JOIN   pg_attribute a ON a.attrelid = i.indrelid
+                                 AND a.attnum = ANY(i.indkey)
+                                 WHERE  i.indrelid = #{quote(quote_table_name(table_name))}::regclass
+                                 AND    i.indisprimary
```

Comment:
Do this query keep the pk column order?
If table definition is `CREATE TABLE comp (a int, b int, c text, d numeric, primary key(b, a));`, it should be dumped as `primary_key: ["b", "a"]`.